### ISSUE TITLE
Port to type parameters (code included)

### ISSUE THREAD

**Issue by kamstrup** [brentp/intintmap#11]
Hi, I wasn't sure exactly how to go about this, so just jumping in

First of all, thanks for an amazing project <3 We have used it a lot at work and the results have been great.

I took a stab at porting everything to type parameters / aka generics, since we have want to use this data structure for uint16s and other integer types as well. I uploaded the result here https://github.com/kamstrup/intmap

I am not sure if it makes sense to create a PR for this project since I broke API and changed prettymuch everything in the effort -- also the intintmap package name no longer applies since only the keys are required to be of int sub-type in the new project, values can be anything. So I renamed to just "intmap" :-)

If you want to update this repo to hold the new project I can def figure something out, just let me know what you think :+1:

**Comment by brentp** [comment]
Hi Mikkel, nice work!
I think it's fine to keep yours as a fork. Would be nice if you had indeed forked the repo rather than creating a new one. 
-Brent

**Comment by kamstrup** [comment]
> Hi Mikkel, nice work! I think it's fine to keep yours as a fork. Would be nice if you had indeed forked the repo rather than creating a new one. -Brent

Yeah, good point. Makes things easier to track. I'll see if I can change it to a fork instead :+1:
